rule.go

```diff
@@ -225,6 +225,7 @@ type Rule struct {
 func (r *Rule) Evaluate(tx *Transaction) []MatchData {
 	if r.Capture {
 		tx.Capture = true
+		defer tx.resetCaptures()
```

Comment:
I think there is a reason why I added r.Capture validation twice, resetCaptures is CPU expensive, so I try to avoid it as much as possible. We must check if that is the case

Reply:
If defer is placed if a for loop, it may cause multiple executions after the function ends.

Reply:
You are right, I think it could be a smarter function like 

```go
defer func (){
 if thereWereMatches { 
   tx.resetCaptures()
  }
}
```

It's a super memory expensive operation.
But let's leave it like this for now.